internal/jobs/twitter.go

```diff
@@ -976,15 +1059,56 @@ func NewTwitterScraper(jc types.JobConfiguration, c *stats.StatsCollector) *Twit
 		configuration:  config,
 		accountManager: accountManager,
 		statsCollector: c,
+		capabilities: map[string]bool{
+			"searchbyquery":       true,
+			"searchbyfullarchive": true,
+			"searchbyprofile":     true,
+			"searchfollowers":     true,
+			"getbyid":             true,
+			"getreplies":          true,
+			"getretweeters":       true,
+			"gettweets":           true,
+			"getmedia":            true,
+			"gethometweets":       true,
+			"getforyoutweets":     true,
+			"getbookmarks":        true,
+			"getprofilebyid":      true,
+			"gettrends":           true,
+			"getfollowing":        true,
+			"getfollowers":        true,
+			"getspace":            true,
+		},
 	}
 }
 
 func (ws *TwitterScraper) ExecuteJob(j types.Job) (types.JobResult, error) {
 	args := &TwitterScraperArgs{}
 	j.Arguments.Unmarshal(args)
+	
+	// Determine which implementation to use based on job type
+	var jobType string = j.Type
+	
 	switch strings.ToLower(args.SearchType) {
 	case "searchbyquery":
-		tweets, err := ws.ScrapeTweetsByRecentSearchQuery(ws.configuration.DataDir, args.Query, args.MaxResults)
+		var tweets []*TweetResult
+		var err error
+		
+		// Select implementation based on job type
+		switch jobType {
```

Comment:
I think this whole thing is screaming for a refactor (something like, use an interface for the different types of scrapes, create structs for each of the types which implement the interface).